### PR TITLE
Support scrolling while dragging

### PR DESCRIPTION
**Pull request by eugeniodepalo**
This PR adds support for scrolling the list container (or any other parent) while the mouse is dragging an item. It does so by remembering what was the offset of the element relative to the document when the drag starts, and by calculating the new offset on every mouse move.

## TODO
- [x] Add a test showcasing the intended behaviour

## Before

p

## After

c

### CODE REVIEW

**Comment by jgwhite** [comment]
Incredible!

**Comment by jgwhite** [comment]
@eugeniodepalo fixing those errors upstream in #40. Once it’s merged you should be good to rebase.

**Comment by jgwhite** [comment]
Merged. Rebase when ready :grin:

**Comment by eugeniodepalo** [comment]
Done! I think this is going to be hard to unit test, maybe an acceptance test makes more sense here - what do you think?

**Comment by jgwhite** [comment]
@eugeniodepalo yes definitely. Or, even better, one of the new-style [component integration tests](http://alisdair.mcdiarmid.org/2015/06/20/ember-component-integration-tests.html).

**Comment by eugeniodepalo** [comment]
Yea, that's what I had in mind. Cool!

File: tests/integration/sortable-group-test.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+function itemHeight() {
+  return $('.sortable-item').outerHeight();
+}
+
+moduleForComponent('sortable-group', 'SortableGroup', { integration: true });
+
+test('Drag an item', function(assert) {
+  $('#ember-testing').css('zoom', '100%');
```
Comment (eugeniodepalo) [inline tests/integration/sortable-group-test.js]: This is a bit of a hack... Without this, the items get adjusted to weird positions while dragging. I assume somewhere there is some positioning code that doesn't account for the zoom.

File: addon/mixins/sortable-item.js [hunk 2]
```diff
@@ -377,17 +384,16 @@ export default Mixin.create({
...
-    return new Promise(resolve => {
-      run.next(() => {
-        let duration = 0;
-
-        if (this.get('isAnimated')) {
-          duration = this.get('transitionDuration');
-        }
-
-        run.later(this, resolve, duration);
+    if (this.get('isAnimated')) {
```
Comment (eugeniodepalo) [inline addon/mixins/sortable-item.js]: This lets the non-async test run smoothly by having the animations disabled. Hopefully this doesn't introduce anything bad.

**Comment by eugeniodepalo** [comment]
@jgwhite added some tests, when you have time to review.

**Comment by jgwhite** [comment]
@eugeniodepalo hey, sorry for the slow response — was on holiday. Will review today.

File: tests/integration/sortable-group-test.js (hunk 1 above)
Comment (jgwhite) [inline tests/integration/sortable-group-test.js]: Yes that’s quite possible. We probably should take zoom into account but that’s an issue for another PR :)

File: addon/mixins/sortable-item.js (hunk 2 above)
Comment (jgwhite) [inline addon/mixins/sortable-item.js]: Maybe I’m reading this wrong, but it looks like the condition is the wrong way around? Shouldn’t it resolve immediately when it *isn’t* animated?

Comment (eugeniodepalo) [inline addon/mixins/sortable-item.js]: Oh wow - not sure how I missed this, it's definitely the other way around.

**Comment by jgwhite** [comment]
@eugeniodepalo I just did a bunch of refactoring. Could you rebase against master?

**Comment by ammorrise** [comment]
Is this on tap to be merged soon?

**Comment by jgwhite** [comment]
@ammorrise waiting for @eugeniodepalo to rebase, then should be good to go.

**Comment by eugeniodepalo** [comment]
Sorry I'm a bit late on this one - was on holiday. I'll rebase and fix the conflicts.

**Comment by jgwhite** [comment]
@eugeniodepalo awesome, thanks!

**Comment by ammorrise** [comment]
@eugeniodepalo Hoping to use this on a project. Do you have an ETA?

**Comment by mehulkar** [comment]
this would be awesome to merge!

**Comment by eugeniodepalo** [comment]
Sorry again for the delay everyone - I lost write access to this repo so I'm going to reopen the PR from my personal fork. I'm hoping to solve the conflicts today.

**Comment by eugeniodepalo** [comment]
Reopened with https://github.com/jgwhite/ember-sortable/pull/58

**Pull request closed without merging**